### PR TITLE
Make DCP the default feed type

### LINKED ISSUE #2388: Make DCP the default feed type
Use DCP as the default feed type for Sync Gateway running in channel cache mode.

Add config validation to throw an error and abort startup if running with FeedType explicitly set to TAP and any of:
 - auto-import set to true
 - XATTR enabled

### PR DESCRIPTION
**Pull request by ajres**
fixes #2388

Rebased against latest master

### CODE REVIEW

File: rest/bulk_api.go [hunk 1]
```diff
@@ -423,7 +423,11 @@ func (h *handler) handleBulkDocs() error {
 			err = base.HTTPErrorf(http.StatusBadRequest, "Document body must be JSON")
 			return err
 		}
-		docid, _ := doc["_id"].(string)
+		docid, ok := doc["_id"].(string)
+		if !ok {
+			err = base.HTTPErrorf(http.StatusBadRequest, "Document id must be string")
+			return err
+		}
```
Comment (adamcfraser) [inline rest/bulk_api.go]: I think this is already on master, isn't it? Should this PR just be rebased on master?

File: base/leaky_bucket.go [hunk 2]
```diff
@@ -283,7 +283,12 @@ func (b *LeakyBucket) UUID() (string, error) {
 		return b.bucket.UUID()
 }
 
-
+func (b *LeakyBucket) CloseAndDelete() error {
+	if bucket, ok := b.bucket.(sgbucket.DeleteableBucket); ok {
+		return bucket.CloseAndDelete()
+	}
+	return nil
+}
```
Comment (adamcfraser) [inline base/leaky_bucket.go:307]: What's changed with this PR to require this change?

Comment (ajres) [reply base/leaky_bucket.go:307]: Originally a failing unit test, failed locally and on drone, can't repro locally now, have pushed with method commented out to see if drone passes as well. Depending on drone result I can remove.

Comment (ajres) [reply base/leaky_bucket.go:307]: Drone failed without this cahnge:
=== RUN   TestFlush
--- FAIL: TestFlush (0.03s)
	api_test.go:254: Response status 503 (expected 200) for POST <http://localhost/db/_flush> : {"error":"Service Unavailable","reason":"Bucket does not support flush"}

Comment (adamcfraser) [reply base/leaky_bucket.go:307]: Ah - didn't notice that the unit tests are always running as walrus wrapped w/ LeakyBucket (for walrus DCP emulation) after your changes to bucket.go.  Makes sense now - thanks.

**Pull request closed without merging**